compile/service_test.go

```diff
@@ -60,53 +60,98 @@ func TestCompileService(t *testing.T) {
 	}
 
 	keyValueSpec := &ServiceSpec{
-		Name: "KeyValue",
-		File: "test.thrift",
+		Name:        "KeyValue",
+		File:        "test.thrift",
+		Annotations: []*AnnotationSpec{},
```

Comment:
the Annotations slice is added even if empty; so the tests fail when comparing []*AnnotationSpec{} with []*AnnotationSpec(nil).

I could potentially not add the slice if len == 0